source/Firmata/UwpFirmata.cpp

```diff
@@ -374,19 +566,56 @@ UwpFirmata::onConnectionEstablished(
 
 void
 UwpFirmata::onConnectionFailed(
-    Platform::String ^message
+    Platform::String ^message_
     )
 {
-    FirmataConnectionFailed( message );
+    FirmataConnectionFailed( message_ );
 }
 
 void
 UwpFirmata::onConnectionLost(
-    Platform::String ^message
+    Platform::String ^message_
     )
 {
     _connection_ready = false;
-    FirmataConnectionLost( message );
+    FirmataConnectionLost( message_ );
+}
+
+void
+UwpFirmata::reassembleByteString(
+    uint8_t *byte_string_,
+    size_t length_
+    )
+{
+    //each char must be reassembled from the two 7-bit bytes received, therefore length should always be an even number.
+    int i, j;
+    for( i = 0, j = 0; j < length_ - 1; ++i, j += 2 )
+    {
+        byte_string_[i] = byte_string_[j] | ( byte_string_[j + 1] << 7 );
+    }
+    byte_string_[i] = 0;
+}
+
+void
+UwpFirmata::sendSysex(
+    uint8_t command_,
+    uint8_t length_,
+    uint16_t *buffer_
+    )
+{
+    //critical section equivalent to function scope
+    std::lock_guard<std::mutex> lock( _firmutex );
+
+    _firmata_stream->write( static_cast<uint8_t>( Command::START_SYSEX ) );
+    _firmata_stream->write( command_ );
+
+    for( uint8_t i = 0; i < length_; ++i )
+    {
+        sendValueAsTwo7bitBytes( buffer_[i] );
```

Comment:
Are all sysEx command blob bytes really expanded this way?  I thought it was message command dependent. For instance: the servo_config command sends the pin number (0-127) as the first byte after the command and then it breaks it apart into lsb and msb pairs.  Would it be better to have the caller of sendSysex handle the encoding of the sysex command blob?